File: Aula Arrays/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int main(void)
{
    string text;
    do
    {
        text = get_string("Insira o texto: ");
    } while(strlen(text) <= 0);


    long letras = 0;
    int palavras = 1;
    int frases = 0;



    for(int i = 0, n = strlen(text); i <= n; i++)
    {
        if(isalpha(text[i]) != 0)
        {
            letras++;
        }
        if(((int)text[i] == 33) || ((int)text[i] == 46) || ((int)text[i] == 63))
        {
            frases++;
        }
        if((int)text[i] == 32)
        {
            palavras++;
        }

    }

    float l;
    float s;
    float grade;

    double averageWordsN = 0;
    averageWordsN = palavras / 100.0;

    l = letras / averageWordsN;
    s = frases / averageWordsN;

    grade = (0.0588 * l) - (0.296 * s) - 15.8;

    if(grade >= 16.0)
    {
        printf(" Letras: %ld\n Frases: %d\n Palavras: %d\n Letras por cada 100 palavras: %.2f\n Frases por cada 100 palavras: %.2f\n Grade: 16+\n", letras, frases, palavras, l, s);
    }
    else if(grade < 1.0)
    {
        printf(" Letras: %ld\n Frases: %d\n Palavras: %d\n Letras por cada 100 palavras: %.2f\n Frases por cada 100 palavras: %.2f\n Before Grade 1\n", letras, frases, palavras, l, s);

    }
    else
    printf(" Letras: %ld\n Frases: %d\n Palavras: %d\n Letras por cada 100 palavras: %.2f\n Frases por cada 100 palavras: %.2f\n Grade: %.0f\n", letras, frases, palavras, l, s, round(grade));


    return 0;
}

File: pset6/readability.py
text = input("Text: ")

wordCounter = 1
charCounter = 0
sentenceCounter = 0

for char in text:
    if ord(char) >= 65 and ord(char) <= 90:
        charCounter += 1
    if ord(char) >= 97 and ord(char) <= 122:
        charCounter += 1


for element in text:
    if element == " ":
        wordCounter += 1
    if element == "!" or element == "." or element == "?":
        sentenceCounter += 1



L = (charCounter / wordCounter) * 100

S = (sentenceCounter / wordCounter) * 100


grade = round((0.0588 * L) - (0.296 * S) - 15.8)

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print("Grade " + str(grade))

